UefiTestingPkg/Include/Library/FlatPageTableLib.h

```diff
@@ -0,0 +1,166 @@
+/** @file
+  Library to parse page/translation table entries.
+
+  Copyright (c) Microsoft Corporation.
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+
+**/
+
+#ifndef FLAT_PAGE_TABLE_LIB_H_
+#define FLAT_PAGE_TABLE_LIB_H_
+
+typedef struct {
+  UINT64    LinearAddress;
+  UINT64    Length;
+  UINT64    PageEntry;
+} PAGE_MAP_ENTRY;
+
+typedef struct {
+  UINT32            ArchSignature;
+  PAGE_MAP_ENTRY    *Entries;
+  UINTN             EntryCount;
+  UINTN             EntryPagesAllocated;
+} PAGE_MAP;
+
+// The signature of the PAGE_MAP struct is used to determine the architecture of the page/translation table
+// entries.
+#define AARCH64_PAGE_MAP_SIGNATURE  SIGNATURE_32 ('A','A','6','4')
+#define X64_PAGE_MAP_SIGNATURE      SIGNATURE_32 ('X','6','4',' ')
+
+// This union can be used to interpret each PAGE_MAP_ENTRY on an AARCH64 system.

```

Comment:
do these bitfields align with page table entry format or anything arch defined or are they just made up with the details you wanted?

Reply:
These align with the table entry format. I'll add a comment noting that and from which architecture revisions they were pulled.

Reply:
Added the architecture revision